### PR TITLE
enable model loading in asset dependencies

### PR DESCRIPTION
**Pull request by franckbrl**
Solves https://github.com/Cornerstone-OnDemand/modelkit/issues/174

### CODE REVIEW

File: modelkit/core/library.py [hunk 1]
```diff
@@ -266,6 +266,8 @@ def _load_model(self, model_name, model_settings=None):
             configuration_key=model_name,
             cache=self.cache,
         )
+        if not self.models[model_name]._loaded:
```
Comment (ldeflandre) [inline modelkit/core/library.py:269]: While it probably fix the issue, I would like to understand why Model are not suffering from the same issue.
If we understand that we will probably find a best place to fix this issue IMHO.

**Review by ldeflandre — CHANGES_REQUESTED** [review]

File: modelkit/core/library.py (hunk 1 above)
Comment (franckbrl) [reply modelkit/core/library.py:269]: Here is the difference between model and asset load methods:
- In `Asset._load`, [only the current asset is loaded](https://github.com/Cornerstone-OnDemand/modelkit/blob/main/modelkit/core/model.py#L148).
- In model loading, [we explicitly go through all dependencies and load them](https://github.com/Cornerstone-OnDemand/modelkit/blob/main/modelkit/core/model.py#L212-L214).

Finally, `ModelLibrary._load_model` does not load dependent models at all, but [only instantiates them](https://github.com/Cornerstone-OnDemand/modelkit/blob/main/modelkit/core/library.py#L259).

Comment (franckbrl) [reply modelkit/core/library.py:269]: So another solution would be to have the same explicit load loop in `Asset._load`, as we saw happends for models.

**Comment by tbascoul** [comment]
Solved by #179

**Pull request closed without merging**